library/java/src/io/envoyproxy/envoymobile/engine/EnvoyNativeResourceRegistry.java

```diff
@@ -0,0 +1,72 @@
+package io.envoyproxy.envoymobile.engine;
+
+import java.lang.ref.PhantomReference;
+import java.lang.ref.ReferenceQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.Set;
+
+import io.envoyproxy.envoymobile.engine.EnvoyNativeResourceReleaser;
+import io.envoyproxy.envoymobile.engine.EnvoyNativeResourceWrapper;
+
+/**
+ * Central class to manage releasing native resources when wrapper objects are flagged as
+ * unreachable by the garbage collector.
+ */
+public enum EnvoyNativeResourceRegistry {
+  SINGLETON;
+
+  private ReferenceQueue
+      refQueue; // References are automatically enqueued when the gc flags them as unreachable.
+  private Set refMaintainer; // Maintains references in the object graph while we wait for them to
+                             // be enqueued.
+  private RefQueueThread
+      refQueueThread; // Blocks on the reference queue and calls the releaser of queued references.
```

Comment:
```suggestion
  // References are automatically enqueued when the gc flags them as unreachable.
  private ReferenceQueue refQueue;

  // Maintains references in the object graph while we wait for them to
  // be enqueued.
  private Set refMaintainer;

  // Blocks on the reference queue and calls the releaser of queued references.
  private RefQueueThread refQueueThread;
```

Reply:
The formatter did this; I didn't even notice. :p